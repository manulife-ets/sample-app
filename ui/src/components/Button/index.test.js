import React from 'react'
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { Button } from '.';

test('Button works', () => {
  const { container } = render(<Button />)
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule('color','#fff');
})

test('Button works with props', () => {
  const { container } = render(<Button bgColor='black' />)
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule('background-color','black');
})
