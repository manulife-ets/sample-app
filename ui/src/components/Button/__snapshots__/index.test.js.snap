// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button works 1`] = `
.c0 {
  min-width: 150px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0;
  text-align: center;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  border: 1px solid transparent;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-weight: 600;
  max-width: 100%;
  white-space: normal;
  background-color: #ec6453;
  color: #fff;
}

.c0:hover {
  background-color: #dc5a44;
}

@media only screen and (max-width :414px) {
  .c0 {
    padding: 1rem 0;
  }
}

<button
  class="c0"
/>
`;

exports[`Button works with props 1`] = `
.c0 {
  min-width: 150px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0;
  text-align: center;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  border: 1px solid transparent;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-weight: 600;
  max-width: 100%;
  white-space: normal;
  background-color: black;
  color: #fff;
}

.c0:hover {
  background-color: #dc5a44;
}

@media only screen and (max-width :414px) {
  .c0 {
    padding: 1rem 0;
  }
}

<button
  class="c0"
/>
`;
